src/cpp/src/vlm_pipeline.cpp

```diff
@@ -0,0 +1,588 @@
+    size_t im_end_id = special_tokens.data<int64_t>()[1];
+    size_t slice_start_id = special_tokens.data<int64_t>()[2];
+    size_t slice_end_id = special_tokens.data<int64_t>()[3];
+    ov::Tensor input_ids = m_tokenizer.encode(new_templated_chat_history).input_ids;
+    m_embedding.set_input_tensor(input_ids);
+    m_embedding.infer();
+    ov::Tensor inputs_embeds = m_embedding.get_output_tensor();
+    OPENVINO_ASSERT(
+        m_vlm_config.hidden_size == inputs_embeds.get_shape().at(2),
+        "Unexpected embedding size"
+    );
+    if (!rgbs.empty()) {
+        int64_t* ids = input_ids.data<int64_t>();
+        const ov::Tensor& resampled_source = resample(*this, embeds.resized_source, {embeds.resized_source_size});
+        float* emb = resampled_source.data<float>();
+        bool replacing = false;
+        for (size_t token_idx = 0; token_idx < inputs_embeds.get_shape().at(1); ++token_idx) {
+            if (im_start_id == ids[token_idx]) {
+                replacing = true;
+            }
+            if (replacing) {
+                std::copy_n(emb, resampled_source.get_size(), inputs_embeds.data<float>() + token_idx * m_vlm_config.hidden_size);
+                token_idx += resampled_source.get_shape().at(1);
+                replacing = false;
+                break;
+            }
+        }
+        if (embeds.slices) {
+            size_t token_idx = 0;
+            const ov::Shape& slices_shape = embeds.slices.get_shape();
+            const std::vector<HeightWidth>& sliced_sizes = embeds.slices_sizes;
+            for (size_t i = 0; i < slices_shape.at(0); ++i) {
+                for (size_t ja = 0; ja < slices_shape.at(1); ++ja) {
+                    size_t d2 = slices_shape.at(2);
+                    size_t d3 = slices_shape.at(3);
+                    ov::Tensor encoded_view{ov::element::f32, {1, d2, d3}, embeds.slices.data<float>() + (i * slices_shape.at(1) + ja) * d2 * d3};
+                    const ov::Tensor& vision_embed_tensor_i_j = resample(*this, encoded_view, {sliced_sizes.at(i * slices_shape.at(1) + ja)});
+                    for (; token_idx < inputs_embeds.get_shape().at(1); ++token_idx) {
+                        if (slice_start_id == ids[token_idx]) {
+                            replacing = true;
+                        }
+                        if (slice_end_id == ids[token_idx]) {
+                            replacing = false;
+                            break;
+                        }
+                        if (replacing) {
+                            std::copy_n(vision_embed_tensor_i_j.data<float>(), vision_embed_tensor_i_j.get_size(), inputs_embeds.data<float>() + token_idx * m_vlm_config.hidden_size);
+                            token_idx += vision_embed_tensor_i_j.get_shape().at(1);
+                            replacing = false;
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
it will be useful for Cont Batching to have functionality like InputEmbedder which is composition of VisionEncoder / Tokenizers and preparing a final `inputs_embeds`, which can be used as input to LLM part, which served by OVMS

Let's consider having such block in future PRs when we will add more pipelines (supposing, that other models have different logic to prepare such `inputs_embeds`)